### ISSUE TITLE
Errors being thrown on data length changes and updates

### ISSUE THREAD

**Issue by vanthiyathevan** [recharts/recharts#24]
First of all Awesome library. I am trying this chart, so far very impressed at the ease of use.

But the following errors
- line chart `Cannot read property 'coord' of undefined` is being thrown when data length increases. Tracked it down to `getComposedData` function in `LineChart.js`. 
- `Cannot read property 'getStyle' of null` is being thrown when data is updated in `handleStyleChange()` function in Animate.js

This is my chart so far
```jsx
<ResponsiveContainer>
  <LineChart data={data} margin={{ top: 15, right: 20, bottom: 5, left: 15 }}>
    <XAxis dataKey="date" strokeOpacity={0}/>
    <YAxis width={10} tickFormatter={() => {}} strokeOpacity={0}/>
    {
      graphs.map(g => (
        <Line
          key={g.name}
          {...g}
          label={<Labeler {...g} lastIndex={data.length - 1}/>}
          dot={g.dot !== false ? <Dotter {...g} /> : false }
        />
      ))
    }
   </LineChart>
</ResponsiveContainer>
```

Thank you

**Comment by xile611** [comment]
Thanks! Can you give the full code in jsfiddle? 
Maybe you can fork one of our demo in <http://recharts.org>

**Comment by vanthiyathevan** [comment]
Here you go http://jsfiddle.net/rezxmfxm/4/

**Comment by xile611** [comment]
The first error is due to forget to update some states when data is update. 
This bug is fixed in version 0.6.3.
The second error is likely due to the props `strokeColors`  in `Dotter` is `undefined`, and the exception seems to be catched in React.

<http://jsfiddle.net/rezxmfxm/5/>

**Comment by vanthiyathevan** [comment]
Thank you
sorry 2nd error was my fault.

**Comment by vanthiyathevan** [comment]
This is working. Thank you. However some of the props doesn't get applied to the graphs such as `strokeDasharray`

**Comment by xile611** [comment]
I find that `strokeDasharray ` doesn't  get applied because we change `strokeDasharray` to generate the animation effect of line. @jasonHzq 
We will consider this problem.

**Comment by xile611** [comment]
Fixed in v0.7.0.
<http://recharts.org/examples#DashedLineChart>
